test/e2e/lib/jest/globalSetup.ts

```diff
@@ -33,6 +39,32 @@ export default async (): Promise< void > => {
 	const userList = [ 'gutenbergSimpleSiteUser', 'eCommerceUser', 'defaultUser' ];
 
 	for await ( const user of userList ) {
+		const cookiePath = path.join( cookieBasePath, `${ user }.json` );
+
+		/*
+		This section is nearly identical to the snippet in `BrowserManager.setLoginCookie`, but
+		instead of checking that cookies are older than 3 days, this checks whether the cookies
+		are newer than 3 days. If so, presumably the cookies are still good to use and the iteration
+		is short-circuited.
+		*/
+		try {
+			await access( cookiePath );
+			const stats = await stat( cookiePath );
+			const createdTime = Math.trunc( stats.birthtimeMs );
+			const currentTime = new Date();
+
+			if ( createdTime > currentTime.setDate( currentTime.getDate() - 3 ) ) {
+				console.log( `\nCookie file on disk for ${ user } is recent, skipping.` );
+				continue;
+			} else {
+				// Remove the stale cookie file as BrowserContext.storageState does not
+				// overwrite files.
+				await unlink( cookiePath );
+			}
+		} catch {
+			// noop
```

Comment:
Should we provide the same feedback [as above](https://github.com/Automattic/wp-calypso/pull/58267/files#diff-1868a0f5bc3c3ac7ed9f7b61df88fc1f401c8b9d97bb413570fdd9738bf31986R200)?

```js
throw new Error( `Cookie file ${ cookiePath } not found on disk.` );
```

Reply:
The error is suppressed for a couple of reasons:

1. if an error is thrown in a `globalSetup` type file, it would block execution of the actual test suite.
2. user who has never run an end to end test locally will not have a cookie file created locally. If `unlink` is called on a path that does not exist, it will throw an exception, and for the reason cited in 1) we do not want to throw an error here.

Reply:
There's also the approach I applied here, which is to make the `globalSetup` as transparent as possible from a user perspective - they should not even realize what is happening and things should just 'work' in a transparent manner, preferably.

Hence my decision to only output the most essential logs only and never to throw an exception in the `globalSetup`.

Reply:
Thanks for elaborating. IMO, it's difficult to say which part of the whole `try` block would throw so in the end it's hard to figure out what we're suppressing. If the whole block depends on the existence of the `cookiePath`, do you think something like this could be more transparent code-wise:

```js
import { access } from 'fs/promises';

// ...

let cookieFileExists = false;
try {
  await access( cookiePath );
  cookieFileExists = true;
} catch {
  // noop
}

if ( cookieFileExists ) {
  // Do things with the cookie file
} else {
  // Do things without the cookie file
}
```

Reply:
I think separating it out like that would allow for a more relevant console output if we deem it necessary, so I am generally on board with the proposal. I'll spin up a new task for now.

Reply:
Node recommends not using `access`, and instead of catching the exception later if the file doesn't exist (when calling `stats` in this case). The rationale an external process could have deleted the file between the call to `access` and the call to `stats`, so you need error control on `stats` anyway.

> Do not use fs.access() to check for the accessibility of a file before calling fs.open(), fs.readFile() or fs.writeFile(). Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file is not accessible.

(From https://nodejs.org/api/fs.html#fsaccesspath-mode-callback)